server/webapp/WEB-INF/rails.new/app/controllers/api_v1/admin/environments_controller.rb

```diff
@@ -0,0 +1,66 @@
+##########################################################################
+# Copyright 2015 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+module ApiV1
+  module Admin
+    class EnvironmentsController < ApiV1::BaseController
+      before_action :check_admin_user_and_401
+      before_action :load_environment, only: [:show, :update, :destroy]
+
+      def index
+        environments = environment_config_service.getEnvironments().to_a
+        render :json_hal_v1 => ApiV1::Config::EnvironmentsConfigRepresenter.new(environments).to_hash(url_builder: self)
+      end
+
+      def show
+        render :json_hal_v1 => ApiV1::Config::EnvironmentConfigRepresenter.new(@environment_config).to_hash(url_builder: self)
+      end
+
+      def create
+        result = HttpLocalizedOperationResult.new
+        environment_config = BasicEnvironmentConfig.new
+        ApiV1::Config::EnvironmentConfigRepresenter.new(environment_config).from_hash(params[:environment])
+        environment_config_service.createEnvironment(environment_config, current_user, result)
+        if result.isSuccessful
+          render :json_hal_v1 => ApiV1::Config::EnvironmentConfigRepresenter.new(environment_config).to_hash(url_builder: self)
+        else
+          json = ApiV1::Config::EnvironmentConfigRepresenter.new(environment_config).to_hash(url_builder: self)
+          render_http_operation_result(result, {data: json})
+        end
+      end
+
+      def update
+        environment_config = BasicEnvironmentConfig.new
+        ApiV1::Config::EnvironmentConfigRepresenter.new(environment_config).from_hash(params[:environment])
+        result = environment_config_service.updateEnvironment(params[:name], environment_config, current_user, params[:md5])
+        render_http_operation_result(result)
+      end
+
+      def destroy
+        result = HttpLocalizedOperationResult.new
+        environment_config_service.deleteEnvironment(params[:name], result)
+        render_http_operation_result(result)
+      end
+
+      private
+      def load_environment(environment_name = params[:name])
```

Comment:
Is there ever a case where you pass a different environment name to have to justify the optional argument?